hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/HoodieSparkTable.java

```diff
@@ -49,8 +48,7 @@
 public abstract class HoodieSparkTable<T extends HoodieRecordPayload>
     extends HoodieTable<T, JavaRDD<HoodieRecord<T>>, JavaRDD<HoodieKey>, JavaRDD<WriteStatus>> {
 
-  private boolean isMetadataAvailabilityUpdated = false;
-  private boolean isMetadataTableAvailable;
+  private volatile boolean isMetadataTableBasePathExists = false;
```

Comment:
nit: can we name this "isMetadataTableExists"

Reply:
+1

Reply:
fixed.